### DISCUSSION TITLE
Changement de nom

### DISCUSSION THREAD

**Discussion by saniho** [saniho/apiEnedis#138] [Ideas]
https://github.com/saniho/apiEnedis/discussions/119#discussioncomment-2254885

**Comment by mdeweerd** [comment]
Je pense que - si on veut laisser du temps aux utilisateurs pour migrer - que ceci esdt une bonne approche:

- Creer un nouveau projet (myenedis ou my_enedis) - pas un fork géré depuis github;
- Pousser le git du présent projet dedans (c'est donc le même contenu, mais github ne comprend pas le lien et le projet n'apparaîtra pas comme un fork);
- Tout futur développement se fait dans ce projet;
- On mets le nouveau projet sous hacs;
- Lors d'un release on pousse les modifications vers "l'ancien" projet, tout en faisant de sorte que l'ancien projet maintient l'ancien domaine et noms de répertoires.
  "Il suffit" de déclencher une action sur la'zncien projet qui déclenche un pull et qui le mets à jour.
- On essaye de voir si on peut récuperer l'ancienne configuration - éventuellement lors de l'initialisation pour que l'utilisateur doit juste confirmer.

A un moment donné, on pourra arrêter de mettre à jour l'ancien projet.

**Reply by saniho** [reply]
je vais faire 2/3 testss de mon coté pour voir ce que l'on pourrait faire ... mais je me dit qu'on peut-etre faire plus simple ....

**Comment by mdeweerd** [comment]
@saniho 
Je pense que le passage à la nouvelle API est un excellent moment pour changer le nom et permettre l'ajout du composant comme un "HACS Default".
Pour éviter que cela se présente comme un fork au sens github, il faut creer le nouveau projet et pousser le clone vers le projet.  Je pense qu'il vaut mieux marquer la différence.
Il sera toujours possible d'avoir du code commun en faisant des merge, etc depuis son PC.
